src/stratigraphy/util/find_description.py

```diff
@@ -59,34 +83,73 @@ def get_description_blocks(
         # vertical spacing between text.
         min_block_count = 2 / 3 * target_layer_count
 
+    count_blocks_divided_by_line = len([block for block in blocks if block.is_terminated_by_line])
     if len(blocks) < min_block_count:
-        # No good splitting by lines found, so create blocks by looking at vertical spacing instead
-        blocks = []
-        current_block_lines = []
-        for line in description_lines:
-            if len(current_block_lines) > 0:
-                last_line = current_block_lines[-1]
-                if (
-                    line.rect.y0 > last_line.rect.y1 + 5
-                    or (  # upper boundary of line is higher up than lower boundary plus 5 points of last line.
-                        threshold and line.rect.y0 > last_line.rect.y1 and line.rect.y0 > last_line.rect.y0 + threshold
-                    )
-                ):
-                    blocks.append(TextBlock(current_block_lines))
-                    current_block_lines = []
-
-            current_block_lines.append(line)
-        if len(current_block_lines):
-            blocks.append(TextBlock(current_block_lines))
-
+        # This case means that there are fewer blocks than the minimum number of blocks we expect.
+        # In this case we redo all the blocks from scratch.
+        blocks = _split_block_by_vertical_spacing(description_lines, threshold=threshold)
+
+    elif count_blocks_divided_by_line < min_block_count:
```

Comment:
That just means, if the blocks are due to lefthandside segments, then still apply vertical distance splitting. But keep the previously made blocks.